server/container/guice/guice-common/src/test/java/org/apache/james/utils/InMemoryMailRepositoryStoreTest.java

```diff
@@ -137,4 +158,25 @@ public void getShouldReturnPreviouslyCreatedMailRepository() throws Exception {
             .contains(mailRepository);
     }
 
+    @Test
+    public void selectShouldNotReturnDifferentResultsWhenUsedInAConcurrentEnvironment() throws Exception {
+        String url = "memory://repo";
+        int threadCount = 10;
+        int operationCount = 1;
+
+        ConcurrentTestRunner concurrentTestRunner = new ConcurrentTestRunner(threadCount, operationCount,
+            (threadNb, operationNb) -> repositoryStore.select(url)
+                .store(FakeMail.builder()
+                    .name("name" + RANDOM.nextInt())
```

Comment:
you don't need random, an AtomicInt would be ok for that IMO

Reply:
Arf it do not hurt right?

If you make a case of this, sure, I'd change it. But why are you against random?

Reply:
Because random is bad (tm).
To explain to others: we try to make our tests as reproducible as possible to avoid random failures.